doc: improve net docs (#32811)
nodejs/node: doc/api/net.md

[`socket.connect(options[, connectListener])`][`socket.connect(options)`]
callback is a listener for the `'connect'` event.

### `socket.destroy([error])`
<!-- YAML
added: v0.1.90
-->

* `error` {Object}
* Returns: {net.Socket}

> Ensures that no more I/O activity happens on this socket. Effectivly discards

Reviewer: ```suggestion
Ensures that no more I/O activity happens on this socket. It effectively discards
```

> any buffered data and ends the socket.

Reviewer 2: This is a bit misleading. Buffered data on the readable side of the socket is not discarded.

Author: Good spot.

Author: Removed.

> the stream but does not explicitly cause a TCP stream reset (RST).

Author: @lpinca: please note updated version, better?

Reviewer 2: I would honestly not add those details that I think are confusing for most users. "Destroys the stream and closes the connection". Interested users can dig by themselves to see what happens under the hood.

Author: Changed to your suggestion.

> See, [`writable.destroy()`][] for further details.

Reviewer: ```suggestion
See [`writable.destroy()`][] for further details.
```

### `socket.destroyed`

* {boolean} Indicates if the connection is destroyed or not. Once a
  connection is destroyed no further data can be transferred using it.

> See, [`writable.destroyed`][] for further details.

Reviewer: ```suggestion
See [`writable.destroyed`][] for further details.
```

Half-closes the socket. i.e., it sends a FIN packet. It is possible the
server will still send some data.

> See, [`writable.end()`][] for further details.

Reviewer: ```suggestion
See [`writable.end()`][] for further details.
```
